### PR TITLE
contract reader api change for block meta data

### PR DESCRIPTION
**Pull request by ettec**
Part of the work for -> https://smartcontract-it.atlassian.net/browse/CAPPL-108

ContractReader API change discussed  here-> https://docs.google.com/document/d/1jyJ2eXIR0EP8eO30Sj4eVj-jJFW34OsnwIzV5mYwKKY/edit?tab=t.0#heading=h.3hiih1q4rib

### CODE REVIEW

File: pkg/types/contract_reader.go [hunk 1]
```diff
@@ -46,6 +50,9 @@ type ContractReader interface {
 	// Passing in a *values.Value as the returnVal will encode the return value as an appropriate value.Value instance.
 	GetLatestValue(ctx context.Context, readIdentifier string, confidenceLevel primitives.ConfidenceLevel, params, returnVal any) error
 
+	// GetLatestValueWithBlockMetaData should be used in the same was as GetLatestValue, but also returns the block metadata.
+	GetLatestValueWithBlockMetaData(ctx context.Context, readIdentifier string, confidenceLevel primitives.ConfidenceLevel, params, returnVal any) (BlockMetaData, error)
```
Comment (krehermann) [inline pkg/types/contract_reader.go:54]: need to add to unimplmented contract reader

File: pkg/types/contract_reader.go [hunk 2]
```diff
@@ -46,6 +50,9 @@ type ContractReader interface {
 	// Passing in a *values.Value as the returnVal will encode the return value as an appropriate value.Value instance.
 	GetLatestValue(ctx context.Context, readIdentifier string, confidenceLevel primitives.ConfidenceLevel, params, returnVal any) error
 
+	// GetLatestValueWithBlockMetaData should be used in the same was as GetLatestValue, but also returns the block metadata.
```
Comment (ilija42) [inline pkg/types/contract_reader.go:53]:
```suggestion
	// GetLatestValueWithBlockMetaData should be used in the same way as GetLatestValue, but also returns the block metadata.
```

File: pkg/types/contract_reader.go [hunk 3]
```diff
@@ -17,6 +17,10 @@ const (
 	ErrNotFound                    = NotFoundError("not found")
 )
 
+type BlockMetaData struct {
+	Height uint64
+}
```
Comment (ettec) [inline pkg/types/contract_reader.go:22]: we can add to this as needed, don't think we need to finalise on what data to expose at this point?

File: pkg/types/contract_reader.go (hunk 1 above)
Comment (ettec) [reply pkg/types/contract_reader.go:54]: ooh, thanks, doing it

File: pkg/types/contract_reader.go [hunk 4]
```diff
@@ -17,6 +17,10 @@ const (
 	ErrNotFound                    = NotFoundError("not found")
 )
 
+type BlockMetaData struct {
```
Comment (ilija42) [inline pkg/types/contract_reader.go:20]: you can reuse Head type that is defined in this same file

File: pkg/types/contract_reader.go (hunk 1 above)
Comment (ettec) [reply pkg/types/contract_reader.go:54]: done, ta

File: pkg/types/contract_reader.go (hunk 2 above)
Comment (ettec) [reply pkg/types/contract_reader.go]: sorted

**Review by ilija42 — APPROVED** [review]

**Review by cedric-cordenier — APPROVED** [review]

**Pull request merged**